src/core/common/tasklet.hpp

```diff
@@ -96,6 +96,44 @@ class Tasklet : public InstanceLocator, public OwnerLocator
     Tasklet *mNext;
 };
 
+/**
+ * This class defines a tasklet which also tracks an arbitrary context information.
```

Comment:
suggestion: "which also tracks an arbitrary context information" -> "that also maintains a user context pointer"